websocket.go

```diff
@@ -85,14 +87,34 @@ func NewWebSocket(si *network.ServerIdentity) *WebSocket {
 	return w
 }
 
+// SetTLS sets the TLSConfig of the server in order to allow TLS communications.
+func (w *WebSocket) SetTLS(tlsCertificate []byte, tlsCertificateKey []byte) error {
+	w.Lock()
+	defer w.Unlock()
+	if w.started {
+		return fmt.Errorf("Cannot set SSL for websocket when it has already been started")
+	}
+	cert, err := tls.X509KeyPair(tlsCertificate, tlsCertificateKey)
+	if err != nil {
+		return err
+	}
+	w.server.Server.TLSConfig = &tls.Config{Certificates: []tls.Certificate{cert}}
```

Comment:
With this API, you make it impossible for the caller to customize the tls.Config if they need to (and we will need to, in order to support LetsEncrypt via the golang.org/x/crypto/acme/autocert package) Another way to do this would be for type WebSocket to have a field called TLSConfig, with a comment saying, "can only be modified before Start is called". Then in app/config.go, instead of calling server.SetWebsockerTLS, you'd set server.WebSocket.TLSConfig as needed. the private field websocket in type onet.Server would need to be exported, but I see no problem with that.

Go prefers correct defaults, and empowering the caller to override them by sending in custom configs. And on balance, a reasonable new variable export (onet.Server.websocket) is better than two new exported APIs.